test/service/test_portal_service.py

```diff
@@ -107,15 +107,14 @@ def tearDown(self):
         super().tearDown()
 
         # To ensure that the bucket is cleared between tests, all versions
-        # must be deleted. The most convenient way to do this is just to
-        # disabling versioning and perform a single delete.
+        # must be deleted.
         self.s3_client.put_bucket_versioning(Bucket=self.portal_service.bucket,
                                              VersioningConfiguration={
                                                  'Status': 'Disabled',
                                                  'MFADelete': 'Disabled'
                                              })
-        self.s3_client.delete_object(Bucket=self.portal_service.bucket,
-                                     Key=self.portal_service.object_key)
+        bucket = aws.resource('s3').Bucket(self.portal_service.bucket)
+        bucket.objects.all().delete()
```

Comment:
If the intent here is to avoid raising an exception if the target object does not exist, surely a simple try/catch would be simpler, clearer, and not run the risk of causing unintended deletions? Otherwise I don't understand the reason for this change.